File: leetcode/anagram.py
# Given two strings s and t, return true if t is an anagram of s, and false otherwise.

# An Anagram is a word or phrase formed by rearranging the letters of a
# different word or phrase, typically using all the original letters exactly
# once.

# I came up with all the below solutions in one attempt. 


from collections import Counter

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         return sorted(s) == sorted(t)

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         d1 = {c: s.count(c) for c in s}
#         d2 = {c: t.count(c) for c in t}
#         return d1 == d2

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         d1 = {}
#         d2 = {}
#         for c in s:
#             d1.setdefault(c, 0)
#             d1[c] += 1
#         for c in t:
#             d2.setdefault(c, 0)
#             d2[c] += 1
#         return d1 == d2

File: leetcode/anagram.js
// Given two strings s and t, return true if t is an anagram of s, and false otherwise.

// An Anagram is a word or phrase formed by rearranging the letters of a
// different word or phrase, typically using all the original letters exactly
// once.

const isAnagram = (s, t) => s.split('').sort().join('') === t.split('').sort().join('')

// const isAnagram = (s, t) => {
//     obj1 = {};
//     obj2 = {};
//     for (let i of s) {
//         if (!(i in obj1)) {
//             obj1[i] = 0;
//         }
//         obj1[i] += 1;
//     }
//     for (let i of t) {
//         if (!(i in obj2)) {
//             obj2[i] = 0;
//         }
//         obj2[i] += 1;
//     }
//     if (Object.keys(obj1).length !== Object.keys(obj2).length) return false;
//     for (let char of Object.keys(obj1)) {
//         if (obj1[char] !== obj2[char]) {
//             return false;
//         }
//     }
//     return true;
// }
